pkg/action/env.container_test.go

```diff
@@ -199,6 +199,65 @@ func Test_ContainerExec_imageEnsure(t *testing.T) {
 	}
 }
 
+func Test_ContainerExec_imageRemove(t *testing.T) {
+	t.Parallel()
+
+	actLoc := testContainerAction(&DefAction{
+		Image: "build:local",
+		Build: &types.BuildDefinition{
+			Context: ".",
+		},
+	})
+	err := actLoc.EnsureLoaded()
+	assert.NoError(t, err)
+	type testCase struct {
+		name     string
+		action   *DefAction
+		expBuild *types.BuildDefinition
+		ret      []interface{}
+	}
+
+	tts := []testCase{
+		{
+			"image removed",
+			actLoc.ActionDef(),
+			nil,
+			[]interface{}{&types.ImageRemoveResponse{Status: types.ImageRemoved}, nil},
+		},
+	}
+
+	for _, tt := range tts {
+		tt := tt
+
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			assert, ctrl, d, r := prepareContainerTestSuite(t)
+			ctx := context.Background()
+
+			defer ctrl.Finish()
+			defer r.driver.Close()
+
+			act := testContainerAction(tt.action)
+			act.input = Input{
+				IO: cli.NoopStreams(),
+			}
+
+			err = act.EnsureLoaded()
+			assert.NoError(err)
+
+			a := act.ActionDef()
+			imgOpts := types.ImageRemoveOptions{Force: false, PruneChildren: false}
+			d.EXPECT().
+				ImageRemove(ctx, a.Image, gomock.Eq(imgOpts)).
+				Return(tt.ret...)
+			res, err := r.driver.ImageRemove(ctx, a.Image, imgOpts)
```

Comment:
You are not testing run environment, you are testing mock, which is not right. 
I suppose you need to write:
```go
d.EXPECT().
  ImageRemove(ctx, a.Image, gomock.Eq(imgOpts)).
  Return(tt.ret...)
res, err := r.imageRemove(ctx, act)
```